### ISSUE TITLE
Dev version in composer.json is pulling in the latest, non-dev version.

### ISSUE THREAD

**Issue by andrewholgate** [drupal-composer/drupal-packagist#28]
Following the instructions to install the [Web Profiler](http://packagist.drupal-composer.org/packages/drupal/webprofiler) a dependency, I add the following to the `composer.json` file:

`"drupal/webprofiler": "8.1.*@dev"`

When I then install via composer, the version which is downloaded appears to be the latest non-dev version (8.1.1-beta9 in this case)

From the downloaded info file (`webprofiler.info.yml`):

```
version: '8.x-1.1-beta9'
core: '8.x'
project: 'webprofiler'
datestamp: 1427355495
```

**Comment by andrewholgate** [comment]
I resolved the issue, I found that in `composer.json` in [Drupal Project](https://github.com/drupal-composer/drupal-project/blob/8.x/composer.json#L32) had been configured to be `"prefer-stable": true`, which will mean it will pull in a stable version if it is available (rather than the dev version).

To force the download of the dev version (even with `prefer-stable` set to `true`), dependencies have to be explicitly declared in the composer.json file.

Eg. They need to be declared as `"drupal/webprofiler": "8.x-1.x-dev"`
Rather than `"drupal/webprofiler": "8.1.*@dev"`
